### ISSUE TITLE
Give up timer on long buffering

### ISSUE THREAD

**Issue by gojzdar** [libre-tube/LibreTube#5474]
### Describe your suggested feature

# Problem:
The player sometimes enters an endless loading mode, when the screen is black and the user has no control over the player, other than closing it.

Rarely, the piped backend fixes the issue and the playback resumes, but it usually takes > 30 minutes, by which point, the user most likely gave up.

# Possible solution
Have a timer from when the last frame was received from the Piped instance and played. If it exceeds a predefined value (maybe 10 minutes), it gives up for good and doesn't try to start loading the video again until the user explicitly requests it.
This state could be indicated with the play button appearing, the same as when the video is paused.

Maybe this would only work when the screen is turned off?
Since if the video doesn't start playing even after 10 minutes, it is safe to assume the user gave up.

# TLDR
Give up on playing the video after N seconds from the last frame received (from Piped instance).

# Priority
Low.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.
